src/test/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferServiceTest.java

```diff
@@ -0,0 +1,226 @@
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, currentTransferStatus);
+
+
+        // when
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId))
+                .thenReturn(conversationRecord);
+
+        transferService.updateConversationTransferStatus(inboundConversationId, newTransferStatus);
+
+        // then
+        verify(transferRepository, never()).updateConversationStatus(any(), any());
+        verify(activityService).concludeConversationActivity(inboundConversationId);
+    }
+
+    @Test
+    void updateConversationTransferStatus_ConversationIsPendingAndNewStatusIsPending_UpdateTransferStatusAndDoNotConcludeConversation() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final ConversationTransferStatus currentTransferStatus = INBOUND_STARTED;
+        final ConversationTransferStatus newTransferStatus = INBOUND_REQUEST_SENT;
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, currentTransferStatus);
+
+        // when
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId))
+                .thenReturn(conversationRecord);
+
+        transferService.updateConversationTransferStatus(inboundConversationId, newTransferStatus);
+
+        // then
+        verify(transferRepository).updateConversationStatus(inboundConversationId, newTransferStatus);
+        verify(activityService, never()).concludeConversationActivity(any());
+    }
+
+    @Test
+    void updateConversationTransferStatus_ConversationIsPendingAndNewStatusIsTerminating_UpdateTransferStatusAndConcludeConversation() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final ConversationTransferStatus currentTransferStatus = INBOUND_REQUEST_SENT;
+        final ConversationTransferStatus newTransferStatus = INBOUND_COMPLETE;
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, currentTransferStatus);
+
+        // when
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId))
+                .thenReturn(conversationRecord);
+
+        transferService.updateConversationTransferStatus(inboundConversationId, newTransferStatus);
+
+        // then
+        verify(transferRepository).updateConversationStatus(inboundConversationId, newTransferStatus);
+        verify(activityService).concludeConversationActivity(inboundConversationId);
+    }
+
+    @Test
+    void updateConversationTransferStatusWithFailure_ConversationIsTerminated_DoNotUpdateTransferStatusToFailedAndConcludeConversation() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final ConversationTransferStatus currentTransferStatus = INBOUND_COMPLETE;
+        final String failureCode = "06";

```

Comment:
I assume there isn't already an enum for these?

Reply:
There is no enum for this at the moment. This'll be something we add when we do the work to send positive/negative acknowledgement messages from ORC